test/integration/two_step_verification_test.rb

```diff
@@ -92,6 +92,17 @@ class TwoStepVerificationTest < ActionDispatch::IntegrationTest
           assert SUCCESS, last_email.subject
         end
       end
+
+      should "accept a valid code from a device which has a small time lag" do
+        old_code = Timecop.freeze(35.seconds.ago) { ROTP::TOTP.new(@new_secret).now }
```

Comment:
I am confident about the code change, but I couldn't totally explain the behaviour I saw when writing the tests:
* this offset would fail reliably with the old implementation and pass with the changes
* When I increased it to 45 seconds, it would fail with the new implementation